backend/src/models/db/tables.js

```diff
@@ -211,8 +229,55 @@ function insertDB (db, table, data, prep) {
     })
 }
 
+// Inserts unique JSON object into database table.
+// data = { column : value }
+// For security reasons, column inputs can NEVER be user defined.
+function insertUniqueDB (db, table, data) {
+    return new Promise((resolve, reject) => {
+        const insertValues = Object.values(data)
+        const insertColumns = Object.keys(data)
+        const insertPlaceholders = insertColumns.map(_ => '?').join()
+        const query = `REPLACE INTO ${table} (${insertColumns})
+        VALUES (${insertPlaceholders})`
+
+        db = db.run(
+            query,
+            [...insertValues],
+            function (err) { err ? reject(err) : resolve(this.lastID) }
+        )
+    })
+}
+
+// Updates given JSON object into database table.
+// data = { column : value }
+// For security reasons, column inputs can NEVER be user defined.
+// TODO - Maybe need a better way todo more complex conditions
+function updateDB (db, table, data, conditions) {
+    return new Promise((resolve, reject) => {
+        const updateValues = Object.values(data)
+        const updateColumns = Object.keys(data)
```

Comment:
```js
const [ columns, values ] = Object.entries(data)
```

Reply:
This didn't work for some reason, didn't waste time looking into it too much (the columns and values are not the same as what the above code produces)